app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt

```diff
@@ -374,23 +389,25 @@ class LinkPreviewDialog : ExtendedBottomSheetDialogFragment(), LinkPreviewErrorV
     }
 
     private fun setPreviewContents(contents: LinkPreviewContents) {
-        if (!contents.extract.isNullOrEmpty()) {
-            binding.linkPreviewExtractWebview.setBackgroundColor(Color.TRANSPARENT)
-            val colorHex = ResourceUtil.colorToCssString(
-                    ResourceUtil.getThemedColor(
-                            requireContext(),
-                            android.R.attr.textColorPrimary
-                    )
-            )
-            val dir = if (L10nUtil.isLangRTL(viewModel.pageTitle.wikiSite.languageCode)) "rtl" else "ltr"
-            binding.linkPreviewExtractWebview.loadDataWithBaseURL(
-                    null,
-                    "${JavaScriptActionHandler.getCssStyles(viewModel.pageTitle.wikiSite)}<div style=\"line-height: 150%; color: #$colorHex\" dir=\"$dir\">${contents.extract}</div>",
-                    "text/html",
-                    "UTF-8",
-                    null
+        binding.linkPreviewExtractWebview.setBackgroundColor(Color.TRANSPARENT)
+        val colorHex = ResourceUtil.colorToCssString(
+            ResourceUtil.getThemedColor(
+                requireContext(),
+                android.R.attr.textColorPrimary
             )
-        }
+        )
+        val dir = if (L10nUtil.isLangRTL(viewModel.pageTitle.wikiSite.languageCode)) "rtl" else "ltr"
+        val editVisibility = (viewModel.fromPlaces || viewModel.fromArticle) && contents.extract.isNullOrEmpty()
```

Comment:
What's the reason for caring if it's `fromPlaces` or `fromArticle`? Don't we want to do the stub treatment in all cases?